src/Lykke.Service.PayInternal/Models/TransferRequestCrosswiseModel.cs

```diff
@@ -0,0 +1,141 @@
+        public string CheckAmountsValidity()
+        {
+            decimal destAmountSum = 0M;
+            foreach (AddressAmount dst in Destinations)
+            {
+                destAmountSum += dst.Amount;
+                if (dst.Amount <= 0)
+                    return "Some destination address has the requested transfer amount <= 0. The transfer is impossible.";
+            }
+
+            decimal sourceAmountSum = 0M;
+            foreach (AddressAmount src in Sources)
+            {
+                sourceAmountSum += src.Amount;
+                if (src.Amount <= 0)
+                    return "Some source address has the transfer amount <= 0. The transfer is impossible.";
+            }
+
+            if (destAmountSum != sourceAmountSum)
+                return "The requested sum for crosswise transfer doesn't coincide with the sum of the specified source addresses' money amounts. The transfer is impossible.";
+
+            return null; // No errors on this level of analysis. But transfer may fail deeper for some reasons.
+        }
+
+        /// <summary>
+        /// Converts the model data to transfer request object with multiple crosswise transactions: each transaction may contain 
+        /// several sources and one destination address, but different transactions (may) have different destination addresses.
+        /// </summary>
+        /// <returns>The newly created instance of <see cref="TransferRequest"/>.</returns>
+        public ITransferRequest ToTransferRequest()
+        {
+            // Please, note: there is no additional call for CheckAmountsValidity(). It is assumed that the caller code already tested it.
+            var result = new TransferRequest()
+            {
+                TransferId = Guid.NewGuid().ToString(),
+                TransferStatus = TransferStatus.InProgress,
+                TransferStatusError = TransferStatusError.NotError,
+                CreateDate = DateTime.Now,
+                MerchantId = MerchantId
+            };
+
+            result.TransactionRequests = new List<ITransactionRequest>();
+            decimal tmpDestinationAmount, iterationAddedAmount;
+            // We need some temporary copy of list of sources for we do iteratively change it below.
+            var tmpSources = (from S in Sources
+                              select new AddressAmount()
+                              {
+                                  Address = S.Address,
+                                  Amount = S.Amount
+                              }).ToList();
+
+            for (int i = 0; i < Destinations.Count; i++)
+            {
+                var newMultipleTransaction = new TransactionRequest();
+                newMultipleTransaction.Amount = Destinations[i].Amount;
+                newMultipleTransaction.CountConfirm = 1; // TODO: maybe it would be more awesome to move this constant to settings
```

Comment:
Didn't get the idea of this property at all.

Reply:
For some reason, it was assumed in legacy code that count of confirmations of a new transaction request is to be set to 1. I've mentioned it above.